components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineViewTest.kt

```diff
@@ -275,4 +276,20 @@ class GeckoEngineViewTest {
         engineView.visibility = View.GONE
         verify(engineView.currentGeckoView)?.visibility = View.GONE
     }
+
+    @Test
+    fun `canClearSelection should return false for null selection or null selection text`() {
+        val engineView = GeckoEngineView(context)
+        engineView.currentGeckoView = mock()
+        engineView.currentSelection = mock()
+
+        // null selection returns false
+        whenever(engineView.currentSelection?.selection).thenReturn(null)
+        Assert.assertFalse(engineView.canClearSelection())
+
+        // null text returns false
```

Comment:
We should also test for the "empty selection text" case. Unfortunately, Mockito can't mock `Selection.text`, but we we're keeping our own subtypes for mocking various GV classes. Here's how this would look in this case:
https://github.com/csadilek/android-components/commit/ffefb0d27052e916c00e25029f84704e00c8edca